### PR TITLE
todo list tweaks and prep for further unit tests

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#5802

*Description of changes:*
- added check for pay-error class
- combined pending status blocks
- replaced array filter with find
- very minor cleanup
- created new test file for template-first tests (very incomplete)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: tests/unit/TodoList2.spec.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
+//
```
Comment (severinbeauvais) [inline tests/unit/TodoList2.spec.ts:1]: As you can see, I got almost nothing implemented here (design only)... and I'm not even happy with the design.

The Todo List component is brutally complicated due to all possible scenarios.

I would like to reorganize the Todo List but I am not sure how we should reorganize it: big code blocks for each possible state? But then there will be a lot of code/HTML duplication.

Clearly, the (partial) day I allocated to implement tests for all scenarios was not enough. I'm open to ideas on how to proceed.

Unfortunately there is no tangible benefit to our users to refactor this. There is only benefit to us in having easier/less fragile code to work on and potentially fewer bugs. But it's going to be a lot of work.

Thoughts?

Comment (severinbeauvais) [reply tests/unit/TodoList2.spec.ts:1]: I could create a new refactor/cleanup ticket but I'm afraid we would never get around to implementing it.

Comment (severinbeauvais) [reply tests/unit/TodoList2.spec.ts:1]: ^^^ I'm going to create a new ticket for this anyways. I think it's our best option atm. (Fixing this now is not in scope.)

... As for this PR, do you think we should merge it?

**Review by vysakh-menon-aot — APPROVED** [review]

File: tests/unit/TodoList2.spec.ts (hunk 1 above)
Comment (cameron-freshworks) [reply tests/unit/TodoList2.spec.ts:1]: Yes to all of the above ( As in i 100% agree :) ). 

Like you said, best to stick with your scope for your sanity sake and we can tackle this when we're able.

**Review by cameron-freshworks — APPROVED** [review]

**Comment by cameron-freshworks** [comment]
Merge whenever your ready!

File: tests/unit/TodoList2.spec.ts (hunk 1 above)
Comment (severinbeauvais) [reply tests/unit/TodoList2.spec.ts:1]: https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/6168

**Pull request merged**